include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -72,20 +64,14 @@ inline Vinecop::Vinecop(const std::vector<size_t>& order,
 //! @brief creates an arbitrary vine copula model.
 //! @param pair_copulas Bicop objects specifying the pair-copulas, see
 //!     make_pair_copula_store().
-//! @param structure an RVineStructure object specifying the vine structure.
+//! @param vine_struct an RVineStructure object specifying the vine structure.
 inline Vinecop::Vinecop(const std::vector<std::vector<Bicop>>& pair_copulas,
-                        const RVineStructure& structure)
+                        const RVineStructure& vine_struct)
+  : Vinecop(vine_struct)
 {
-
-  d_ = structure.get_dim();
-  vine_struct_ = structure;
-
   check_pair_copulas_rvine_structure(pair_copulas);
-
   pair_copulas_ = pair_copulas;
   vine_struct_.truncate(pair_copulas.size());
-  threshold_ = 0.0;
-  loglik_ = NAN;
```

Comment:
Should we maybe set the loglik if the pair copulas have already a loglik?

Reply:
Hm not sure, the vine object hasn't seen any data...